etna/models/nn/deepar_new.py

```diff
@@ -0,0 +1,383 @@
+from collections import Counter
+from typing import Any
+from typing import Dict
+from typing import Iterator
+from typing import Optional
+from typing import Type
+
+import numpy as np
+import pandas as pd
+from typing_extensions import TypedDict
+
+from etna import SETTINGS
+from etna.distributions import BaseDistribution
+from etna.distributions import FloatDistribution
+from etna.distributions import IntDistribution
+from etna.models.base import DeepBaseModel
+from etna.models.base import DeepBaseNet
+
+if SETTINGS.torch_required:
+    import torch
+    import torch.nn as nn
+    from torch.distributions import NegativeBinomial
+    from torch.distributions import Normal
+    from torch.utils.data.sampler import Sampler
+
+
+class DeepARSampler(Sampler):
```

Comment:
Better to name it according to the strategy, in fact we can use [this sampler](https://pytorch.org/docs/stable/data.html#torch.utils.data.WeightedRandomSampler) -- we need to create wrapper which construct this sampler inside getting the weights from the `data`

Reply:
Just add notes that this sampling method was used in the paper